packages/sanity/src/core/studio/copyPaste/CopyPasteProvider.tsx

```diff
@@ -32,14 +63,285 @@ export const CopyPasteProvider: React.FC<{
+        // needs the document and absolute path to the field
+        targetRootPath: targetPath,
+        targetRootValue: value,
+        options: {
+          validateAssets: true,
+          validateReferences: true,
+          client,
+        } as TransferValueOptions,
+      }
+      copiedJsonTypes.push(sourceSchemaType.jsonType)
+
+      try {
+        const {targetValue, errors} = await transferValue(transferValueOptions)
+        const nonWarningErrors = errors.filter((error) => error.level !== 'warning')
+        const _isEmptyValue = isEmptyValue(targetValue)
+
+        if (nonWarningErrors.length > 0) {
+          const description = t(nonWarningErrors[0].i18n.key, nonWarningErrors[0].i18n.args)
+
+          toast.push({
+            status: 'error',
+            title: t('copy-paste.on-paste.validation.clipboard-invalid.title'),
+            description,
+          })
+          return
+        }
+
+        if (errors.length > 0 && !_isEmptyValue) {
+          const description = errors.map((error) => t(error.i18n.key, error.i18n.args)).join(', ')
+
+          toast.push({
+            status: 'warning',
+            title: t('copy-paste.on-paste.validation.partial-warning.title'),
+            description,
+          })
+        }
+
+        if (_isEmptyValue) {
+          toast.push({
+            status: 'warning',
+            title: t('copy-paste.on-paste.validation.clipboard-empty.title'),
+          })
+          return
+        }
+
+        updateItems.push({patches: [set(targetValue, targetPath)], targetSchemaTypeTitle})
+      } catch (error) {
+        toast.push({
+          status: 'error',
+          title: error.message,
+        })
+        return
+      }
+
+      telemetry.log(FieldPasted, {
+        context: options?.context?.source || 'unknown',
+        schemaTypes: copiedJsonTypes,
+      })
+
+      if (updateItems.length) {
+        const allPatches = flatten(updateItems.map(({patches}) => patches))
+        const allTargetNames = updateItems.map((i) => i.targetSchemaTypeTitle).join('", "')
+
+        if (!documentMeta?.onChange) {
+          console.warn(`Failed to resolve document onChange method when pasting.`)
+
+          toast.push({
+            status: 'error',
+            title: t('copy-paste.on-copy.validation.document-metadata-unknown-error.title'),
+          })
+          return
+        }
+
+        documentMeta.onChange(PatchEvent.from(allPatches))
+
+        if (clipboardItem.isDocument) {
+          toast.push({
+            status: 'success',
+            title: t('copy-paste.on-paste.validation.document-paste-success.title', {
+              fieldNames: allTargetNames,
+            }),
+          })
+
+          return
+        }
+
+        const isSingleField = updateItems.length === 1
+
+        if (isSingleField) {
+          toast.push({
+            status: 'success',
+            title: t('copy-paste.on-paste.validation.field_one-paste-success.title', {
```

Comment:
super nit: is the snake case here a typo?

Reply:
This is a convention in i18next to allow for [singular/plural combinations](https://www.i18next.com/translation-function/plurals#singular-plural). I mostly copied these over from the previous place these were.